unitTests/ui/CategoryFacet/CategoryFacetTest.ts

```diff
@@ -27,6 +27,105 @@ export function CategoryFacetTest() {
     let test: IBasicComponentSetup<CategoryFacet>;
     let simulateQueryData: ISimulateQueryData;
 
+    describe('exposes options', () => {
+      describe('dependsOn', () => {
```

Comment:
Maybe those "depends on" tests should be moved to a new DependentFacetManagerTest.ts file where you could test for both facet types, and then remove them here and in FacetTest.ts

Reply:
Agreed, I think it would be more natural.

Reply:
I agree with Jean-Felix on centralising the tests in a new file rather than duplicating them. The only test(s) we should have in the CategoryFacet and Facet files are checks to make sure that the correct methods on `DependsOnManager` are called. That is, we test the "interfaces", not the core logic.

e.g.
```
when triggering `DeferredQuerySuccess`, it calls the `updateVisibilityBasedOnDependsOn` on the DependsOnManager.
```

One thing that might be difficult is spying on the method? Let me know if you have difficulties there.

Reply:
👍  Ok I'll move it to a new file and delete it from the original file

@samisayegh 
I was thinking of doing something like this.
In the test, rebind a new function on the `deferredQuerySucces`, where I create my own dependsOn manager and then spy on it to see if it is called. I would not call `removeFading` like in the original function but I don't think it's important for this test.

Reply:
Hmm, if we bind a different function, we would only be testing the event emitter aspect rather than the DependsOnManager method.

Is it possible to do:
```
spyOn(updateVisibilityBasedOnDependsOn, facet['dependsOnManager'])
```
I think this would work even if `dependsOnManager` is a private property?

Then, after triggering the event, we would expect `facet['dependsOnManager'].updateVisibilityBasedOnDependsOn` to have been called 1 time.

Reply:
I think I'll work. 
I saw somewhere somebody called a private function with that way
I'll try that.

Reply:
👍  works perfectly